lib/util/http/cache/abstract-http-cache-provider.ts

```diff
@@ -43,17 +46,35 @@ export abstract class AbstractHttpCacheProvider implements HttpCacheProvider {
     return Promise.resolve(null);
   }
 
+  private preventCaching<T>(resp: HttpResponse<T>): boolean {
+    if (this.checkCacheControlPublic === false) {
+      return false;
+    }
+
+    const isPublic = resp.headers?.['cache-control']
+      ?.toLocaleLowerCase()
+      ?.split(regEx(/\s*,\s*/))
+      ?.includes('public');
```

Comment:
```suggestion
      .split(regEx(/\s*,\s*/))
      .includes('public');
```
I think it's safe without